### PR TITLE
Add com.github.powertab.powertabeditor

### PR DESCRIPTION
**Pull request by cameronwhite**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - `--device=all` is needed for MIDI output (via [RtMidi](https://www.music.mcgill.ca/~gary/rtmidi/))
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
  - I am the upstream maintainer at https://github.com/powertab/powertabeditor
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.powertab.powertabeditor.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+finish-args:
+  # Standard Qt stuff.
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  # For MIDI devices
+  - --device=all
+
+modules:
+  - shared-modules/linux-audio/jack2.json
```
Comment (hfiguiere) [inline com.github.powertab.powertabeditor.yml:18]: JACK needs at least `--filesystem=xdg-run/pipewire-0` in the finish args for audio playback.

File: com.github.powertab.powertabeditor.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    sources:
+      - type: git
+        url: https://github.com/thestk/rtmidi.git
+        tag: 5.0.0
+        commit: 84a99422a3faf1ab417fe71c0903a48debb9376a
+
+  - name: powertabeditor
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.github.powertab.powertabeditor.yml:76]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

File: com.github.powertab.powertabeditor.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    sources:
+      - type: git
+        url: https://github.com/zeux/pugixml.git
+        tag: v1.11.4
+        commit: 08b3433180727ea2f78fe02e860a08471db1e03c
+
+  - name: nlohmann-json
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.github.powertab.powertabeditor.yml:55]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

File: com.github.powertab.powertabeditor.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    sources:
+      - type: git
+        url: https://github.com/nlohmann/json.git
+        tag: v3.10.4
+        commit: fec56a1a16c6e1c1b1f4e116a20e79398282626c
+
+  - name: rtmidi
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.github.powertab.powertabeditor.yml:66]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

File: com.github.powertab.powertabeditor.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        sha256: 4ff941449631ace0d4d203e3483be9dbc9da454084111f97ea0a2114e19bf066
+    build-commands:
+      - autoreconf -fi
+      - ./configure --prefix=/app
+      - make install
+
+  - name: pugixml
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.github.powertab.powertabeditor.yml:45]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.github.powertab.powertabeditor

**Comment by hfiguiere** [comment]
the appid in the appdata is incorrect.

File: com.github.powertab.powertabeditor.yml (hunk 1 above)
Comment (cameronwhite) [reply com.github.powertab.powertabeditor.yml:20]: Good catch, thanks! My testing so far was just with ALSA

**Comment by cameronwhite** [comment]
bot, build com.github.powertab.powertabeditor

**Comment by cameronwhite** [comment]
bot, build com.github.powertab.powertabeditor

**Comment by cameronwhite** [comment]
Thanks for the feedback! I've addressed those comments and fixed a couple subsequent build errors.

**Comment by cameronwhite** [comment]
bot, build com.github.powertab.powertabeditor

**Comment by barthalion** [comment]
bot, build com.github.powertab.powertabeditor

**Pull request closed without merging**